woodwork/column_schema.py

```diff
@@ -10,40 +10,53 @@
 from woodwork.utils import _convert_input_to_set
 
 
-def _get_column_dict(name=None,
-                     logical_type=None,
-                     semantic_tags=None,
-                     use_standard_tags=False,
-                     description=None,
-                     metadata=None,
-                     validate=True):
-    """Creates a dictionary that contains the typing information for a column schema
-    Args:
-        name (str, optional): The name of the column.
-        logical_type (str, LogicalType, optional): The column's LogicalType.
-        semantic_tags (str, list, set, optional): The semantic tag(s) specified for the column.
-        use_standard_tags (boolean, optional): If True, will add standard semantic tags to the column based
-                on the specified logical type if a logical type is defined for the column. Defaults to False.
-        description (str, optional): User description of the column.
-        metadata (dict[str -> json serializable], optional): Extra metadata provided by the user.
-        validate (bool, optional): Whether to perform parameter validation. Defaults to True.
-    """
-    if metadata is None:
-        metadata = {}
-    if validate:
-        if logical_type is not None:
-            _validate_logical_type(logical_type)
-        _validate_description(description)
-        _validate_metadata(metadata)
-
-    semantic_tags = _get_column_tags(semantic_tags, logical_type, use_standard_tags, name, validate)
-
-    return {
-        'logical_type': logical_type,
-        'semantic_tags': semantic_tags,
-        'description': description,
-        'metadata': metadata
-    }
+class ColumnSchema(object):
+    def __init__(self,
+                 name=None,
+                 logical_type=None,
+                 semantic_tags=None,
+                 use_standard_tags=False,
+                 description=None,
+                 metadata=None,
+                 validate=True):
+        """Create ColumnSchema
+
+        Args:
+            name (str, optional): The name of the column.
```

Comment:
Yeah, I see your point here. I'd be fine removing it--won't be a breaking change, I think, because users didn't have access to `_get_column_dict` before, and we already handle the null name value in `_convert_input_to_set`. 

Plus, we will catch this error in the TableSchema because we convert the inputs to set before initializing individual `ColumnSchema`s so if someone had this issue in a table, they'd still get the good error with the column name. I see no problem in removing it!